### PR TITLE
move snatch_parameters to normbase, add running mean/var and num batc…

### PR DESCRIPTION
**Pull request by nixli**
…hes tracked stats to snatch_parameters, update test accordingly:

* [ yes ] Tested: integration test, made sure the test breaks without the change and passes with the change
* [ yes ] Formatted with YAPF

### CODE REVIEW

File: hfta/ops/batchnorm_test.py [hunk 1]
```diff
@@ -55,6 +55,44 @@ def testcase_1d(
...
+    for i in range(train_test_steps):
+      if L == 0:
+        x_array = [torch.rand(N, C) for _ in range(B)]
+      else:
+        x_array = [torch.rand(N, C, L) for _ in range(B)]
+
+      _ = [batchNormal1d_array[b](x_array[b]) for b in range(B)]
+
+    batchNormal1d_fused.reset_parameters()
+    # snatch the "trained" BN layer
```
Comment (wangshangsam) [inline hfta/ops/batchnorm_test.py:69]: Hmmm, there's neither backward passes nor parameter update steps, how is this `# emulate "training" for the BN layer`?

**Review by wangshangsam — CHANGES_REQUESTED** [review]

File: hfta/ops/batchnorm_test.py (hunk 1 above)
Comment (nixli) [reply hfta/ops/batchnorm_test.py:69]: I thought running forward pass already changes the internal state of BN layer ?
the weights and bias

Comment (wangshangsam) [reply hfta/ops/batchnorm_test.py]: Yes, sorry, you are right, my bad :)
`running_mean`, `running_var` and `num_batches_tracked` are registered as buffers: https://github.com/UofT-EcoSystem/hfta/blob/main/hfta/ops/batchnorm.py#L47, and indeed they are not updated by the optimizers: https://discuss.pytorch.org/t/what-is-the-difference-between-register-buffer-and-register-parameter-of-nn-module/32723 . I misread them as `register_parameter()`

**Review by wangshangsam — CHANGES_REQUESTED** [review]
I'm taking a look at the unit test again ...

It seems that your only purpose is to make sure `snatch_parameters` works as expected on `running_mean`, `running_var` and `num_batches_tracked`? If that's the case, I would just initialize all the parameters and buffers of unfused ops to some random values (cuz the fact that they are initialized to constant values by default is the reason why we weren't able to catch this error in the first place), and see, after `snatch_parameters`, if the fused op produces the same result on the same input. Actually training those unfused BatchNorm ops to get different values for the parameters and buffers seems overkill to me. 

That being said, I do think that (as a huge piece of engineering work) we need to test both the forward pass and the backward pass in our unit tests (currently we only make sure the forward pass produces the same result); however, this might be something that needs to be addressed in a separate PR for all ops, instead of in this PR.

**Comment by nixli** [comment]
that sounds good! let me make the change

**Review by wangshangsam — APPROVED** [review]
LGTM!
